### PR TITLE
protonmail-bridge: 1.1.5 -> 1.1.6

### PR DESCRIPTION
**Pull request by GRBurst**
###### Motivation for this change
Update: 1.1.5 -> 1.1.6

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg build protonmail-bridge

**Review by mmahut — APPROVED** [review]
- [X] reviewed the diff and commit messages
- [X] made sure ofBorg build succeeded for all applicable platforms
- [X] run nix-review without any failures
- [X] run and tested the binaries

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 1]
```diff
@@ -67,6 +78,13 @@ in stdenv.mkDerivation rec {
       $out/lib/protonmail-bridge
   '';
 
+  postInstall = ''
```
Comment (worldofpeace) [inline pkgs/applications/networking/protonmail-bridge/default.nix]: We have qt wrappers now so we don't need to roll our own anymore.

Please read
* https://hydra.nixos.org/build/96749518/download/1/nixpkgs/manual.html#sec-language-qt

to see how to adapt the expression to use it.

File: pkgs/applications/networking/protonmail-bridge/default.nix [hunk 2]
```diff
@@ -67,6 +78,13 @@ in stdenv.mkDerivation rec {
...
+    wrapProgram $out/bin/spectral \
+      --set QML2_IMPORT_PATH "${qml2ImportPath}"
```
Comment (worldofpeace) [inline pkgs/applications/networking/protonmail-bridge/default.nix]: Why is this needed when we're using `wrapQtAppsHook`? The issue was most likely caused by none of the qt deps being in `buildInputs` so the hook couldn't collect anything.

Also you should switch this expression to use `mkDerivation` which will automatically add `wrapQtAppsHook`.

Comment (GRBurst) [reply pkgs/applications/networking/protonmail-bridge/default.nix]: ah yeah - forgot to cleanup. my bad 👍

**Comment by worldofpeace** [comment]
This comment

> Also you should switch this expression to use `mkDerivation` which will automatically add wrapQtAppsHook.
> to nativeBuildInputs without explicitly specifying this

still stands.

**Comment by GRBurst** [comment]
> Also you should switch this expression to use `mkDerivation` which will automatically add wrapQtAppsHook.
> to nativeBuildInputs without explicitly specifying this

I don't get that exactly. The package build with a `mkDerivation` expression (line 24). What am I missing?

**Comment by GRBurst** [comment]
> > Also you should switch this expression to use `mkDerivation` which will automatically add wrapQtAppsHook.
> > to nativeBuildInputs without explicitly specifying this
> 
> I don't get that exactly. The package is build with a `mkDerivation` expression (line 24). What am I missing?

Oh - I guess I misunderstood your comment.

**Comment by worldofpeace** [comment]
`libsForQt5.callPackage` brings is `qt5` in scope so it would be `qt5.mkDerivation`.

so `stdenv.mkDerivation` != `mkDerivation` in this case.

Understand the confusion, it makes the expression appear rather ambiguous.

**Comment by GRBurst** [comment]
> `libsForQt5.callPackage` brings is `qt5` in scope so it would be `qt5.mkDerivation`.
> 
> so `stdenv.mkDerivation` != `mkDerivation` in this case.
> 
> Understand the confusion, it makes the expression appear rather ambiguous.

Thank you for your guidance :+1:

**Comment by worldofpeace** [comment]
But `libsForQt5.callPackage` brings `qt5.mkDerivation` in scope so you must reference it as just `mkDerivation`.

Same with `qtbase` etc.

**Comment by GRBurst** [comment]
If I just reference it as just `mkDerivation` this does not work.
```sh
error: undefined variable 'mkDerivation' at /nixpkgs/pkgs/applications/networking/protonmail-bridge/default.nix:24:4
```
Hence I added `qt5` such that I can call `qt5.mkDerivation`, which worked (compiles and binary runs).

**Comment by worldofpeace** [comment]
You'd need to add it to the function arguments
```
{ ... , mkDerivation }:
```

**Review by worldofpeace — APPROVED** [review]
Builds and executes for me locally.

**Comment by worldofpeace** [comment]
After merging this I noticed that the archive includes a desktop file that's more complete than the one we generate.

```
[Desktop Entry]
Name=ProtonMail Bridge
GenericName=ProtonMail Bridge for Linux
Comment=The Bridge is an application that runs on your computer in the background and seamlessly encrypts and decrypts your mail as it enters and leaves your computer.
Exec=protonmail-bridge
Icon=/usr/share/icons/protonmail/ProtonMail_Bridge.svg
Type=Application
Version=1.0
Terminal=false
Categories=Utility;Security;Network;Email
```

Though those paths would need to be patched with `substitute`.
